sdk/nodejs/npm/manager.go

```diff
@@ -0,0 +1,107 @@
+package npm
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+)
+
+// A `PackageManager` is responsible for installing dependencies,
+// packaging Pulumi programs, and executing Node in the context of
+// installed packages. In practice, each implementation of this
+// interface represents one of the package managers in the Node ecosystem
+// e.g. Yarn, NPM, etc.
+// The language host will dynamically dispatch to an instance of PackageManager
+// in response to RPC requests.
+type PackageManager interface {
+	Install(ctx context.Context, dir string, production bool, stdout, stderr io.Writer) error
+	Pack(ctx context.Context, dir string, stderr io.Writer) ([]byte, error)
+}
+
+// Pack runs `npm pack` in the given directory, packaging the Node.js app located there into a
+// tarball and returning it as `[]byte`. `stdout` is ignored for the command, as it does not
+// generate useful data. If the `PULUMI_PREFER_YARN` environment variable is set, `yarn pack` is run
+// instead of `npm pack`.
+func Pack(ctx context.Context, dir string, stderr io.Writer) ([]byte, error) {
```

Comment:
I wouldn't worry too much about this, just keep the interface as is. It will all need to change at some point as we move "packing" over the grpc boundary into language plugins, and that work is fairly iniment.